app/src/main/java/org/elasticsearch/node/internal/CrateSettingsPreparer.java

```diff
@@ -150,17 +150,21 @@ private static String randomNodeName() {
 
         try {
             List<String> names = new ArrayList<>();
-            try (BufferedReader reader = new BufferedReader(new InputStreamReader(input, Charsets.UTF_8))) {
-                String line = reader.readLine();
-                while (line != null) {
-                    String[] fields = line.split("\t");
-                    if (fields.length == 0) {
-                        throw new RuntimeException("Failed to parse the names.txt. Malformed record: " + line);
+            try (InputStreamReader streamReader = new InputStreamReader(input, Charsets.UTF_8)) {
+                try (BufferedReader reader = new BufferedReader(streamReader)) {
```

Comment:
afaik the `BufferedReader` closes it's argument on close itself, so splitting the try blocks shouldn't be necessary

Reply:
you're right. wasn't aware of that.